crates/ruff_linter/src/rules/ruff/rules/tuple_map_int_version_parsing.rs

```diff
@@ -0,0 +1,161 @@
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{Expr, ExprAttribute, ExprCall, ExprName};
+use ruff_python_semantic::SemanticModel;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for calls of the form `tuple(map(int, __version__.split(".")))`.
+///
+/// ## Why is this bad?
+/// `__version__` does not always contain integral-like elements.
+///
+/// ```python
+/// import matplotlib  # 3.9.1.post-1
+///
+/// # ValueError: invalid literal for int() with base 10: 'post1'
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// See also [PEP 440].
+///
+/// ## Example
+/// ```python
+/// tuple(map(int, matplotlib.__version__.split(".")))
+/// ```
+///
+/// Use instead:
+/// ```python
+/// import packaging.version as version
+///
+/// version.parse(matplotlib.__version__)
+/// ```
+///
+/// [PEP 440]: https://peps.python.org/pep-0440/
+#[violation]
+pub struct TupleMapIntVersionParsing;
+
+impl Violation for TupleMapIntVersionParsing {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "`__version__` may contain non-integral-like elements".to_string()
+    }
+}
+
+/// RUF048
+pub(crate) fn tuple_map_int_version_parsing(checker: &mut Checker, call: &ExprCall) {
+    let semantic = checker.semantic();
+
+    let Some(Expr::Call(child_call)) = tuple_like_call_with_single_argument(semantic, call) else {
+        return;
+    };
+
+    let Some((first, second)) = map_call_with_two_arguments(semantic, child_call) else {
+        return;
+    };
+
+    if !semantic.match_builtin_expr(first, "int") || !is_dunder_version_split_dot(second) {
+        return;
+    }
+
+    let diagnostic = Diagnostic::new(TupleMapIntVersionParsing, call.range());
+
+    checker.diagnostics.push(diagnostic);
+}
+
+fn tuple_like_call_with_single_argument<'a>(
+    semantic: &SemanticModel,
+    call: &'a ExprCall,
+) -> Option<&'a Expr> {
+    let (func, positionals) = func_and_positionals(call)?;
+    let func_is = |symbol: &str| semantic.match_builtin_expr(func, symbol);
+
+    if !func_is("tuple") && !func_is("list") || positionals.len() != 1 {
+        return None;
+    };
+
+    positionals.first()
+}
+
+fn map_call_with_two_arguments<'a>(
+    semantic: &SemanticModel,
+    call: &'a ExprCall,
+) -> Option<(&'a Expr, &'a Expr)> {
+    let (func, positionals) = func_and_positionals(call)?;
+
+    if !semantic.match_builtin_expr(func, "map") || positionals.len() != 2 {
+        return None;
+    };
+
+    Some((positionals.first().unwrap(), positionals.last().unwrap()))
```

Comment:
It's generally good to avoid `.unwrap()` calls where possible. Even in cases like this where you can see that they're safe, it's a sign that you're fighting the type system rather than using it to your advantage. Here you can do this to achieve the same thing without any `.unwrap()`s.

```suggestion
    if !semantic.match_builtin_expr(func, "map") {
        return None;
    };

    if let [first, second] = positionals {
        Some((first, second))
    } else {
        None
    }
```